### DISCUSSION TITLE
How to specify `font-color` for inline text?

### DISCUSSION THREAD

**Discussion by mingsu** [quarto-dev/quarto-cli#3496] [Q&A]
For example:

```qmd

I want the some words in [yellow]{.yellow}, but some words in [red]{.red}.

Expected to work in `docx`, `pdf` and `html` formats.

```

**Comment by mcanouil** [comment]
You need to explain a bit more.
If you have some colour definition for example in a CSS file, your example is exactly what you need to colour some words or do whatever you want.

**Reply by mingsu** [reply]
Thanks for your reply.

I understand that my syntax should work on `html` format, if I set corresponding `css classes`. But I wonder it to work not just `html` format, but also `pdf` and `docx` format.

For `pdf` format, I can use quarto extension [`latex-environment`](https://github.com/quarto-ext/latex-environment) to set the colored words by define new `environment` (e.g. `colword`), and use it  like this:

```qmd
I want the some words in [yellow]{colword, options="yellow"}, but some words in [red]{colword, options="red"}.
```

But I expect a more universal solution, so that works over multiple output format.

**Reply by dragonstyle** [reply]
I don't know of a general solution that exists yet, though that would be a cool extension. It could emit the proper elements to convert spans with color attributes / classes to the appropriate structure for the output (very much like `latex-environment` but perhaps doing a bit more work as far as processing the span to implement the color in the various output formats).

**Reply by TobGerken** [reply]
This thread is old, but I am trying my luck. I am writing a report using quarto and while I can get the latex-environment extension to work for a single qmd document. When adding the same header to the book, it will fail to generate colored text. When I look at the .tex that is generated, I can see that no color is applied. 
Do you know the reason why this might fail when using a report?

**Reply by mcanouil** [reply]
May I suggest another approach? Use the following extension and simply ignore the format.

- https://github.com/mcanouil/quarto-highlight-text


Extensions listing: https://m.canouil.dev/quarto-extensions/

**Reply by TobGerken** [reply]
Thanks that worked, but updating to quarto 1.7 broke my table of contents, sigh

**Comment by tomkom** [comment]
I have stumbled upon this thread, and face the same problem.

I need to underline a text with a dashed (or dotted) line.

I have a project that has a pdf, html, and revealjs output. But for now, I cannot even get it to run in HTML.

I have made a new css (output_style.css), and referenced it in the theme in my _quarto.yml:

```
format:
    html:
        theme: [cosmo,output_style.css]
```

in the css, I added:

```
.udot {
  text-decoration-line: underline;
  text-decoration-color: rgb(50, 50, 50);
  text-decoration-style: dashed;
  text-decoration-thickness: 1px;
}
```

And in my doc.qmd, I tried:

this [text]{.udot} should be underlined dotted.

No luck... What am I missing ( I am no css wizard :) ).

**Reply by mcanouil** [reply]
For short css is not scss as described in <https://quarto.org/docs/reference/formats/html.html#format-options>.

The correct/proper approach:
<img width="1624" alt="image" src="https://user-images.githubusercontent.com/8896044/225020681-fca24f6f-e2ba-42e7-a581-ef4a327439e6.png">

Trick Quarto into thinking css is scss but not quite (not recommended at all):
<img width="1624" alt="image" src="https://user-images.githubusercontent.com/8896044/225023883-def5e366-11e9-4ef8-a2c5-50976d8ba19f.png">

**Reply by tomkom** [reply]
That worked, thank you!
I did not know about the option to add the css: key in the yaml, along with theme. That did not seem to be documented in the reference.

**Reply by tomkom** [reply]
And now it is there loud and clear :)
